test/sequential/test-fs-watch-system-limit.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+const stream = require('stream');
+
+const processes = [];
+const gatherStderr = new stream.PassThrough();
+gatherStderr.setEncoding('utf8');
+gatherStderr.setMaxListeners(Infinity);
+
+let finished = false;
+function spawnProcesses() {
+  for (let i = 0; i < 10; ++i) {
+    const proc = child_process.spawn(
+      process.execPath,
+      [ '-e',
+        `process.chdir(${JSON.stringify(__dirname)});
+        for (const file of fs.readdirSync('.'))
+          fs.watch(file, () => {});`
+      ], { stdio: ['inherit', 'inherit', 'pipe'] });
+    proc.stderr.pipe(gatherStderr);
+    processes.push(proc);
+  }
+
+  setTimeout(() => {
+    if (!finished && processes.length < 200)
```

Comment:
Does the test fails for timeout if `process.length` is 200 and `'Error:'` is never written to `gatherStderr`?

Reply:
Yes, I assume so. The idea is to just skip the test on platforms where that happens … does that sound reasonable?

Reply:
yes